src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1993,7 +1993,8 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
         }
     }
 
-    bool is_initial_query = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::INITIAL_QUERY;
+    bool is_initial_query = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::INITIAL_QUERY ||
+                            (getContext()->getZooKeeperMetadataTransaction() && getContext()->getZooKeeperMetadataTransaction()->isInitialQuery());
```

Comment:
One small detail I forgot about in my previous PR is that Initial queries in Database Replicated have query_kind == ClientInfo::QueryKind::SECONDARY_QUERY at this point in the code, and we need to check for getZooKeeperMetadataTransaction() to find it out.

Reply:
Let's add a comment, as it looks a bit counter-intuitive given variable name.